`pinocchio-log` compile error during build `--arch v3 `

### Problem
compilation of `pinocchio-log` fails when attempt to build for target `sbpfv3-solana-solana`

error:
```bash
error[E0603]: function `sol_log_` is private
   --> src/logger.rs:141:19
    |
141 |         syscalls::sol_log_(message.as_ptr(), message.len() as u64);
    |                   ^^^^^^^^ private function
    |
note: the function `sol_log_` is defined here
   --> src/logger.rs:16:5
    |
16  |     unsafe extern "C" fn sol_log_(message: *const u8, length: u64) {
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0603`.
error: could not compile `pinocchio-log` (lib) due to 1 previous error
```

build command :
```bash
cargo-build-sbf --verbose --arch v3 ...
```

solana version:
```bash
solana-cli 2.2.15 (src:00000000; feat:798020478, client:Agave)
```

cargo-build-sbf version:
```bash
solana-cargo-build-sbf 2.2.15
platform-tools v1.48
rustc 1.84.1
```

### Potential Solution

Export `sol_log_` from syscalls, like follows:

```diff
-unsafe extern "C" fn sol_log_(message: *const u8, length: u64) {
-      let syscall: extern "C" fn(*const u8, u64) = core::mem::transmute(544561597u64); // murmur32 hash of "sol_log_"
-      syscall(message, length)
- }

+pub(crate) fn sol_log_(message: *const u8, length: u64) {
+       let syscall: extern "C" fn(*const u8, u64) = unsafe{ core::mem::transmute(544561597u64) }; // murmur32 hash of "sol_log_"
+       syscall(message, length)
+ }
```

--- a/sdk/log/crate/src/logger.rs
+++ b/sdk/log/crate/src/logger.rs
@@ -14,7 +14,7 @@ mod syscalls {
 mod syscalls {
     // Syscalls provided by the SVM runtime (SBPFv3 and newer).
     pub(crate) fn sol_log_(message: *const u8, length: u64) {
-        let syscall: extern "C" fn(*const u8, u64) = unsafe{ core::mem::transmute(544561597u64) }; // murmur32 hash of "sol_log_"
+        let syscall: extern "C" fn(*const u8, u64) = unsafe { core::mem::transmute(544561597u64) }; // murmur32 hash of "sol_log_"
         syscall(message, length)
     }
 

--- a/sdk/log/crate/src/logger.rs
+++ b/sdk/log/crate/src/logger.rs
@@ -13,8 +13,8 @@ mod syscalls {
 #[cfg(all(target_os = "solana", target_feature = "static-syscalls"))]
 mod syscalls {
     // Syscalls provided by the SVM runtime (SBPFv3 and newer).
-    unsafe extern "C" fn sol_log_(message: *const u8, length: u64) {
-        let syscall: extern "C" fn(*const u8, u64) = core::mem::transmute(544561597u64); // murmur32 hash of "sol_log_"
+    pub(crate) fn sol_log_(message: *const u8, length: u64) {
+        let syscall: extern "C" fn(*const u8, u64) = unsafe{ core::mem::transmute(544561597u64) }; // murmur32 hash of "sol_log_"
         syscall(message, length)
     }
 

